lib/python/picongpu/picmi/simulation.py

```diff
@@ -100,56 +179,10 @@ def __yee_compute_cfl_or_delta_t(self) -> None:
             # if neither delta_t nor cfl are given simply silently pass
             # (might change in the future)
 
-    def __init__(
-        self,
-        picongpu_template_dir: typing.Optional[typing.Union[str, pathlib.Path]] = None,
-        picongpu_typical_ppc: typing.Optional[int] = None,
-        picongpu_moving_window_move_point: typing.Optional[float] = None,
-        picongpu_moving_window_stop_iteration: typing.Optional[int] = None,
-        **kw,
-    ):
-        # delegate additional work to parent
-        super().__init__(**kw)
-
-        # additional checks on inputs, @todo move to picmistandard, Brian Marre, 2024
-
-        ## throw if both cfl & delta_t are set
-        if self.solver is not None and "Yee" == self.solver.method and isinstance(self.solver.grid, Cartesian3DGrid):
-            self.__yee_compute_cfl_or_delta_t()
-
-        # store picongpu specific stuff
-        # @todo switch to pydantic for automatic instrumentation of init method, Brian Marre 2024
-        self.picongpu_typical_ppc = picongpu_typical_ppc
-
-        # internal stuff for PICMI interface only
-        self.__runner = None
-        self.__electron_species = None
-        self.__picongpu_custom_input = None
-
-        # set PyPIConGPU template directory
-        if picongpu_template_dir is None:
-            self.picongpu_template_dir = None
-        else:
-            assert "" != picongpu_template_dir, "picongpu_template_dir MUST NOT be empty"
-            # note: pathlib.Path(pathlib.Path(...)) is valid
-            template_path = pathlib.Path(picongpu_template_dir)
-            assert template_path.is_dir(), "picongpu_template_dir must be existing dir"
-            self.picongpu_template_dir = str(template_path)
-
-        self.moving_window_move_point = picongpu_moving_window_move_point
-        self.moving_window_stop_iteration = picongpu_moving_window_stop_iteration
-
-        self.picongpu_typical_ppc = picongpu_typical_ppc
-
-        # store runner state
-        self.__runner = None
-
-        self.__electron_species = None
-
     def __get_operations_simple_density(
         self,
-        pypicongpu_by_picmi_species: typing.Dict[picmistandard.PICMI_Species, species.Species],
-    ) -> typing.List[species.operation.SimpleDensity]:
+        pypicongpu_by_picmi_species: typing.Dict[Species, pypicongpu.species.Species],
+    ) -> typing.List[pypicongpu.species.operation.SimpleDensity]:
```

Comment:
Sorry - this is just the general renaming that was introduced with this PR as well. You can keep it in the ionization PR. (Just started to get into reviewing your PR again.)

Reply:
It actually a little bit more than a renaming.

I changed from using the PICMI species class to mandating PIConGPU inheriting PICMI interface species objects since we rely on methods only present in the daughter class.